### PR TITLE
Rename manifest_version to manifest-version

### PR DESCRIPTION
**Pull request by alexcrichton**
The current manifests encode this with a dash in the name, so we should preserve
that!

### CODE REVIEW

File: src/tools/build-manifest/src/main.rs [hunk 1]
```diff
@@ -171,7 +170,12 @@ impl Builder {
         self.cargo_version = self.version("cargo", "x86_64-unknown-linux-gnu");
 
         self.digest_and_sign();
-        let manifest = self.build_manifest();
+        let Manifest { manifest_version, date, pkg } = self.build_manifest();
+        let mut manifest = BTreeMap::new();
+        manifest.insert("manifest-version".to_string(),
```
Comment (brson) [inline src/tools/build-manifest/src/main.rs]: Please add a comment explaining that rebuilding of the manifest BTreeMap is to correct the 'manifest-version' name.

**Comment by brson** [comment]
r=me w/ nit

**Comment by alexcrichton** [comment]
Nominating for a beta backport

**Pull request merged**